cdap-ui/app/cdap/components/FieldLevelLineage/v2/Context/FllContextHelper.ts

```diff
@@ -14,61 +14,119 @@
  * the License.
 */
 
-// Parses an incoming or outgoing entity object from backend response to get unique nodes (one per incoming or outgoing field), connections, and an object with fieldnames for each incoming or outgoing dataset
+// types for backend response
+interface IFllEntity {
+  entityId?: IEntityId;
+  relations?: IRelation[];
+}
+
+interface IEntityId {
+  namespace?: string;
+  dataset?: string;
+  entity?: string;
+}
+
+export interface IRelation {
+  source?: string;
+  destination?: string;
+}
+
+// These types are used by the frontend
+export interface IField {
+  id: string;
+  type: string;
+  name: string;
+  dataset: string;
+  namespace: string;
+}
 
-export function parseRelations(namespace, target, ents, isCause = true) {
-  const tables = {};
-  const relNodes = [];
+export interface ILink {
+  source: IField;
+  destination: IField;
+}
+
+export interface ITableFields {
+  [tablename: string]: IField[];
+}
+
+// Parses an incoming or outgoing entity object from backend response
```

Comment:
nit: Make this multiline comment. Would be helpful in vscode to show the function description in the intelliSense tooltip.